src/MagicBall/MagicBall.js

```diff
@@ -0,0 +1,85 @@
+    constructor(props) {
+        super(props)
+        this.state = {
+            title: 'Магический шар предсказаний',
+            answers: [
+                { color: 'blue', answer: 'Бесспорно' },
+                { color: 'blue', answer: 'Предрешено' },
+                { color: 'blue', answer: 'Никаких сомнений' },
+                { color: 'blue', answer: 'Определённо да' },
+                { color: 'blue', answer: 'Можешь быть уверен в этом' },
+                { color: 'green', answer: 'Мне кажется - "да"' },
+                { color: 'green', answer: 'Вероятнее всего' },
+                { color: 'green', answer: 'Хорошие перспективы' },
+                { color: 'green', answer: 'Знаки говорят  - "да"' },
+                { color: 'green', answer: 'Да' },
+                { color: 'yellow', answer: 'Пока не ясно, попробуй ещё' },
+                { color: 'yellow', answer: 'Спроси позже' },
+                { color: 'yellow', answer: 'Лучше не рассказывать' },
+                { color: 'yellow', answer: 'Сейчас нельзя предсказать' },
+                { color: 'yellow', answer: 'Сконцентрируйся и спроси опять' },
+                { color: 'red', answer: 'Даже не думай' },
+                { color: 'red', answer: 'Мой ответ - "нет"' },
+                { color: 'red', answer: 'По моим данным - "нет"' },
+                { color: 'red', answer: 'Перспективы не очень хорошие' },
+                { color: 'red', answer: 'Весьма сомнительно' },
+            ],
+            ballWrapper: 'Узнай свою судьбу',
+            inputValue: '',
+            historyWrapper: [],
+        }
+    }
+    handlerOnclick = () => {
+        if (this.state.inputValue) {
+            const random = Math.floor(Math.random() * this.state.answers.length)
+            const randomAnswer = this.state.answers[random]
+            const newHistory = { question: this.state.inputValue, answer: randomAnswer.answer, color: randomAnswer.color }
+            const historyWrapper = [newHistory, ...this.state.historyWrapper]
+            this.setState({
+                ballWrapper: randomAnswer.answer,
+                historyWrapper: historyWrapper,
+                inputValue: '',
+            })
+        }
+    }
+    handlerOnChange = (event) => {
+        this.setState({
+            inputValue: event.target.value
+        })
+    }
+    render() {
+        return (
+            <>
+                <h1 style={{ textAlign: 'center' }}>{this.state.title}</h1>
+                <div className='inlineFlex'>
+                    <InputWrapper value={this.state.inputValue} onChange={this.handlerOnChange} />
+                    <BallWrapper answer={this.state.ballWrapper} onClick={this.handlerOnclick} />
+                    <div>
+                        <h3 style={{ textAlign: 'center' }}>Ваши предсказания:</h3>
+                        <div className='border' >
+                            {this.state.historyWrapper ? this.state.historyWrapper.map((elem, i) => {
```

Comment:
you can use next syntax:

this.state?.historyWrapper.?map

https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Operators/Optional_chaining